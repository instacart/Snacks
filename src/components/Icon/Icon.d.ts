import * as React from 'react'
import { Interpolation } from '@emotion/core'
import hexValues from './hexValues'

export interface IconProps {
  className?: string

  /** String name of icon - ex 'cart' */
  name?: keyof typeof hexValues

  /** Hexcode of desired icon from ic-icons */
  code?: string

  /**
   * Optional style overrides merged into emotion css
   *
   * @deprecated
   * This prop exists for backwards compatibility and will be
   * removed in a future version
   */
  style?: Interpolation

  /** Callback function called after button click */
  onClick?: React.ComponentProps<'i'>['onClick']
}

declare const Icon: React.ComponentType<IconProps>

export default Icon
