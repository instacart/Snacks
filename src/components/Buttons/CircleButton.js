/**
 * A simple circular button
 *
 * @author [Dominic Cocchiarella ](https://github.com/docchia)
 */
import React from 'react'
import PropTypes from 'prop-types'
import responsive from '../../styles/responsive'
import zIndex from '../../styles/zIndex'
import { spacings } from '../../styles/spacing'
import withTheme from '../../styles/themer/withTheme'
import { themePropTypes } from '../../styles/themer/utils'

const styles = {
  default: {
    ...zIndex.Z_INDEX1,
    border: '0',
    width: `${spacings.XL}px`,
    height: `${spacings.XL}px`,
    borderRadius: '50%',
    boxShadow: '0 1px 2px 0 rgba(0, 0, 0, 0.26), 0 1px 4px 0 rgba(0, 0, 0, 0.16)',
    transition: 'background-color 150ms ease-in-out',
    textAlign: 'center',
    lineHeight: '1',
    [responsive.xs]: {
      display: 'none',
    },
  },
}

const CircleButton = props => {
  const { action, actionHover, primaryBackground } = props.snacksTheme.colors

  return (
    <button
      className={props.className}
      onClick={e => {
        if (props.onClick) {
          e.preventDefault()
          props.onClick()
        }
      }}
      aria-label={props.ariaLabel}
      css={[
        styles.default,
        {
          backgroundColor: action,
          color: primaryBackground,
          ':hover': {
            backgroundColor: actionHover,
          },
          ':active': {
            backgroundColor: actionHover,
          },
          ':focus': {
            backgroundColor: actionHover,
          },
        },
        props.style,
      ]}
      disabled={props.disabled}
    >
      {props.children}
    </button>
  )
}

CircleButton.propTypes = {
  /** Label to be used by screen readers */
  ariaLabel: PropTypes.string,
  children: PropTypes.node,
  className: PropTypes.string,
  /** Bool to disable click/touch listeners */
  disabled: PropTypes.bool,
  /** Callback function called after button click */
  onClick: PropTypes.func,
  /** snacks theme attributes */
  snacksTheme: themePropTypes,
  /**
   * Optional style overrides merged into emotion css
   *
   * @deprecated
   * This prop exists for backwards compatibility and will be
   * removed in a future version
   */
  style: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
}

export default withTheme(CircleButton)
