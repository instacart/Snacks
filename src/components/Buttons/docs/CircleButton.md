CircleButton example:
    <div>
      <CircleButton>+</CircleButton>
    </div>
