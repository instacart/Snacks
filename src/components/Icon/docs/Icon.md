Icon example:
```jsx
<div>
  <Icon name={'cart'} />
  <Icon code={'e08e'} />
  <Icon
    name={'likeFilled'}
    style={{
      backgroundColor: '#43B02A',
      color: '#fff',
      borderRadius: '50%',
      padding: '5px'
    }}
  />
</div>
```

See full list of available icons under the Icons section.
