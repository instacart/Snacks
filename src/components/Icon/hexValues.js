export default {
  accountFilled: 'e044',
  account: 'e043',
  addToOrderAndroid: 'e092',
  addToOrderFilled: 'e093',
  addToOrder: 'e094',
  adjustment: 'e087',
  alcoholFilled: 'e049',
  alcohol: 'e01b',
  arrowDownSmallBold: 'e013',
  arrowDownSmall: 'e014',
  arrowEnter: 'e04a',
  arrowLeftSmallBold: 'e011',
  arrowLeftSmall: 'e012',
  arrowLeft: 'e04b',
  arrowRightSmallBold: 'e00f',
  arrowRightSmall: 'e010',
  arrowRight: 'e027',
  arrowUpSmallBold: 'e00d',
  arrowUpSmall: 'e00e',
  bagFilled: 'e038',
  bag: 'e037',
  bellFilled: 'e047',
  bell: 'e01a',
  bogoFilled: 'e097',
  bogo: 'e02c',
  browseFilled: 'e00c',
  browse: 'e09b',
  calendar: 'e015',
  carFilled: 'e01c',
  car: 'e019',
  cartFilled: 'e02d',
  cart: 'e023',
  categories: 'e0a9',
  checkBold: 'e03c',
  checkProgressComplete: 'e088',
  checkProgressDashed: 'e089',
  check: 'e03b',
  clockFilled: 'e04c',
  clock: 'e017',
  compassFilled: 'e046',
  creditCardFilled: 'e04d',
  creditCard: 'e048',
  dealsFilled: 'e05e',
  dealsIcon: 'e07b',
  deals: 'e04e',
  emailFilled: 'e02f',
  email: 'e084',
  facebookFilled: 'e030',
  faq: 'e031',
  filterAndroid: 'e076',
  filterFilled: 'e078',
  filter: 'e077',
  flagFilled: 'e098',
  flag: 'e08c',
  gearFilled: 'e09c',
  gear: 'e099',
  gift: 'e09e',
  giftcardFilled: 'e0a1',
  giftcard: 'e09f',
  googleFilled: 'e050',
  google: 'e04f',
  gridFilled: 'e08b',
  grid: 'e08a',
  guaranteedFreshFilled: 'e054',
  guaranteedFresh: 'e055',
  hamburger: 'e02e',
  happy: 'e029',
  helpFilled: 'e053',
  help: 'e00b',
  homeFilled: 'e05f',
  home: 'e022',
  iconExit: 'e08d',
  iconPersonAddFilled: 'e090',
  iconPersonAdd: 'e08f',
  iconPersonFilled: 'e091',
  iconPerson: 'e08e',
  inSeasonFilled: 'e079',
  inSeason: 'e070',
  infoFilled: 'e052',
  info: 'e001',
  instagram: 'e073',
  items: 'e0a2',
  itemsFilled: 'e06f',
  itemsYouveOrdered: 'e0a3',
  lightbulbAndroid: 'e07d',
  lightbulbFilled: 'e07e',
  lightbulb: 'e07c',
  likeFilled: 'e081',
  like: 'e080',
  listsAndRecipesFilled: 'e07f',
  listsAndRecipes: 'e002',
  listsAdd: 'e045',
  locationMarkerFilled: 'e051',
  locationMarker: 'e003',
  lock: 'e018',
  logout: 'e004',
  loyaltyCardFilled: 'e056',
  loyaltyCard: 'e021',
  messageFilled: 'e042',
  message: 'e041',
  minusBold: 'e058',
  minus: 'e057',
  money: 'e005',
  moreFilled: 'e059',
  more: 'e006',
  noFilled: 'e03f',
  no: 'e040',
  noteFilled: 'e061',
  note: 'e060',
  officeFilled: 'e063',
  office: 'e062',
  orderHistory: 'e007',
  orderProblem: 'e064',
  orderReview: 'e071',
  organicFilled: 'e05b',
  organic: 'e05a',
  phoneAndroidFilled: 'e066',
  phoneAndroid: 'e065',
  phoneFilled: 'e03d',
  phoneIphoneFilled: 'e06a',
  phoneIphone: 'e06b',
  phone: 'e03e',
  picture: 'e024',
  pinterestFilled: 'e05c',
  plusBold: 'e02b',
  plus: 'e02a',
  popular: 'e072',
  pricingFilled: 'e01e',
  pricingSameFilled: 'e020',
  pricingSame: 'e01f',
  pricing: 'e01d',
  quickAdd: 'e0a6',
  reactionAdd: 'e0a0',
  receipt: 'e028',
  reccuring: 'e0a7',
  refund: 'e085',
  replaceBold: 'e06c',
  replace: 'e067',
  request: 'e086',
  save: 'e0a8',
  saveFilled: 'e0aa',
  scissors: 'e074',
  search: 'e008',
  shareAndroid2Filled: 'e03a',
  shareAndroid2: 'e039',
  shareAndroidFilled: 'e032',
  shareAndroid: 'e033',
  sort: 'e09d',
  specialsFilled: 'e068',
  specials: 'e069',
  starFilled: 'e026',
  star: 'e025',
  storeFilled: 'e075',
  store: 'e009',
  thumbsDown: 'e096',
  thumbsUp: 'e095',
  ticketFilled: 'e036',
  ticket: 'e00a',
  tipFilled: 'e05d',
  tip: 'e016',
  tobacco: 'e035',
  trashFilled: 'e0a5',
  trash: 'e0a4',
  twitterFilled: 'e034',
  viewFilled: 'e083',
  view: 'e082',
  weight: 'e09a',
  xBold: 'e07a',
  xCircleFilled: 'e06e',
  xCircle: 'e06d',
  x: 'e000'
}
