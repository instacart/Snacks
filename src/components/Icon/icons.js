export {default as accessibility} from './icons/accessibility.svg'
export {default as accessibilityFilled} from './icons/accessibilityFilled.svg'
export {default as account} from './icons/account.svg'
export {default as accountFilled} from './icons/accountFilled.svg'
export {default as addToOrder} from './icons/addToOrder.svg'
export {default as addToOrderFilled} from './icons/addToOrderFilled.svg'
export {default as adjustment} from './icons/adjustment.svg'
export {default as alcohol} from './icons/alcohol.svg'
export {default as alcoholFilled} from './icons/alcoholFilled.svg'
export {default as appStore} from './icons/appStore.svg'
export {default as arrowDown} from './icons/arrowDown.svg'
export {default as arrowDownSmall} from './icons/arrowDownSmall.svg'
export {default as arrowLeft} from './icons/arrowLeft.svg'
export {default as arrowLeftSmall} from './icons/arrowLeftSmall.svg'
export {default as arrowRight} from './icons/arrowRight.svg'
export {default as arrowRightSmall} from './icons/arrowRightSmall.svg'
export {default as arrowUp} from './icons/arrowUp.svg'
export {default as arrowUpSmall} from './icons/arrowUpSmall.svg'
export {default as backgroundCheck} from './icons/backgroundCheck.svg'
export {default as bag} from './icons/bag.svg'
export {default as bagFilled} from './icons/bagFilled.svg'
export {default as barcode} from './icons/barcode.svg'
export {default as bell} from './icons/bell.svg'
export {default as bellFilled} from './icons/bellFilled.svg'
export {default as browse} from './icons/browse.svg'
export {default as browseFilled} from './icons/browseFilled.svg'
export {default as calendar} from './icons/calendar.svg'
export {default as call} from './icons/call.svg'
export {default as callFilled} from './icons/callFilled.svg'
export {default as car} from './icons/car.svg'
export {default as carFilled} from './icons/carFilled.svg'
export {default as cart} from './icons/cart.svg'
export {default as cartFilled} from './icons/cartFilled.svg'
export {default as categories} from './icons/categories.svg'
export {default as categoriesFilled} from './icons/categoriesFilled.svg'
export {default as chat} from './icons/chat.svg'
export {default as chatFilled} from './icons/chatFilled.svg'
export {default as checkmark} from './icons/checkmark.svg'
export {default as checkmarkCircle} from './icons/checkmarkCircle.svg'
export {default as checkmarkCircleDashed} from './icons/checkmarkCircleDashed.svg'
export {default as checkmarkCircleFilled} from './icons/checkmarkCircleFilled.svg'
export {default as clock} from './icons/clock.svg'
export {default as clockFilled} from './icons/clockFilled.svg'
export {default as coffee} from './icons/coffee.svg'
export {default as cold} from './icons/cold.svg'
export {default as copy} from './icons/copy.svg'
export {default as coupons} from './icons/coupons.svg'
export {default as couponsFilled} from './icons/couponsFilled.svg'
export {default as creditCard} from './icons/creditCard.svg'
export {default as delete} from './icons/delete.svg'
export {default as deleteFilled} from './icons/deleteFilled.svg'
export {default as email} from './icons/email.svg'
export {default as emailFilled} from './icons/emailFilled.svg'
export {default as facebook} from './icons/facebook.svg'
export {default as filter} from './icons/filter.svg'
export {default as fire} from './icons/fire.svg'
export {default as fireFilled} from './icons/fireFilled.svg'
export {default as flag} from './icons/flag.svg'
export {default as flagFilled} from './icons/flagFilled.svg'
export {default as gear} from './icons/gear.svg'
export {default as gearFilled} from './icons/gearFilled.svg'
export {default as gift} from './icons/gift.svg'
export {default as giftcard} from './icons/giftcard.svg'
export {default as glitter} from './icons/glitter.svg'
export {default as glitterFilled} from './icons/glitterFilled.svg'
export {default as google} from './icons/google.svg'
export {default as graph} from './icons/graph.svg'
export {default as graphFilled} from './icons/graphFilled.svg'
export {default as help} from './icons/help.svg'
export {default as helpFilled} from './icons/helpFilled.svg'
export {default as home} from './icons/home.svg'
export {default as homeFilled} from './icons/homeFilled.svg'
export {default as inSeason} from './icons/inSeason.svg'
export {default as inSeasonFilled} from './icons/inSeasonFilled.svg'
export {default as info} from './icons/info.svg'
export {default as infoFilled} from './icons/infoFilled.svg'
export {default as instagram} from './icons/instagram.svg'
export {default as instructions} from './icons/instructions.svg'
export {default as instructionsFilled} from './icons/instructionsFilled.svg'
export {default as items} from './icons/items.svg'
export {default as itemsFilled} from './icons/itemsFilled.svg'
export {default as lightbulb} from './icons/lightbulb.svg'
export {default as lightbulbFilled} from './icons/lightbulbFilled.svg'
export {default as lightning} from './icons/lightning.svg'
export {default as lightningFilled} from './icons/lightningFilled.svg'
export {default as like} from './icons/like.svg'
export {default as likeFilled} from './icons/likeFilled.svg'
export {default as listsAdd} from './icons/listsAdd.svg'
export {default as locationCurrent} from './icons/locationCurrent.svg'
export {default as locationCurrentFilled} from './icons/locationCurrentFilled.svg'
export {default as locationMarker} from './icons/locationMarker.svg'
export {default as locationMarkerFilled} from './icons/locationMarkerFilled.svg'
export {default as lock} from './icons/lock.svg'
export {default as lockFilled} from './icons/lockFilled.svg'
export {default as logout} from './icons/logout.svg'
export {default as loyaltyCard} from './icons/loyaltyCard.svg'
export {default as loyaltyCardFilled} from './icons/loyaltyCardFilled.svg'
export {default as menu} from './icons/menu.svg'
export {default as minus} from './icons/minus.svg'
export {default as money} from './icons/money.svg'
export {default as moneyFilled} from './icons/moneyFilled.svg'
export {default as moon} from './icons/moon.svg'
export {default as moonFilled} from './icons/moonFilled.svg'
export {default as more} from './icons/more.svg'
export {default as moreFilled} from './icons/moreFilled.svg'
export {default as no} from './icons/no.svg'
export {default as noFilled} from './icons/noFilled.svg'
export {default as office} from './icons/office.svg'
export {default as officeFilled} from './icons/officeFilled.svg'
export {default as order} from './icons/order.svg'
export {default as orderIssue} from './icons/orderIssue.svg'
export {default as orderIssueFilled} from './icons/orderIssueFilled.svg'
export {default as orderFilled} from './icons/orderFilled.svg'
export {default as orderReview} from './icons/orderReview.svg'
export {default as orderReviewFilled} from './icons/orderReviewFilled.svg'
export {default as orders} from './icons/orders.svg'
export {default as ordersFilled} from './icons/ordersFilled.svg'
export {default as organic} from './icons/organic.svg'
export {default as organicFilled} from './icons/organicFilled.svg'
export {default as pencil} from './icons/pencil.svg'
export {default as pencilFilled} from './icons/pencilFilled.svg'
export {default as person} from './icons/person.svg'
export {default as personAdd} from './icons/personAdd.svg'
export {default as personAddFilled} from './icons/personAddFilled.svg'
export {default as personFilled} from './icons/personFilled.svg'
export {default as phone} from './icons/phone.svg'
export {default as phoneFilled} from './icons/phoneFilled.svg'
export {default as photo} from './icons/photo.svg'
export {default as photoMissing} from './icons/photoMissing.svg'
export {default as pinterest} from './icons/pinterest.svg'
export {default as playStore} from './icons/playStore.svg'
export {default as plus} from './icons/plus.svg'
export {default as reaction} from './icons/reaction.svg'
export {default as reactionAdd} from './icons/reactionAdd.svg'
export {default as recipes} from './icons/recipes.svg'
export {default as recurringOrder} from './icons/recurringOrder.svg'
export {default as recurringOrderFilled} from './icons/recurringOrderFilled.svg'
export {default as refund} from './icons/refund.svg'
export {default as reorder} from './icons/reorder.svg'
export {default as reorderFilled} from './icons/reorderFilled.svg'
export {default as replacement} from './icons/replacement.svg'
export {default as sale} from './icons/sale.svg'
export {default as saleFilled} from './icons/saleFilled.svg'
export {default as saveForLater} from './icons/saveForLater.svg'
export {default as saveForLaterFilled} from './icons/saveForLaterFilled.svg'
export {default as scan} from './icons/scan.svg'
export {default as scissors} from './icons/scissors.svg'
export {default as search} from './icons/search.svg'
export {default as sendEmail} from './icons/sendEmail.svg'
export {default as sendEmailFilled} from './icons/sendEmailFilled.svg'
export {default as share} from './icons/share.svg'
export {default as shareFilled} from './icons/shareFilled.svg'
export {default as skip} from './icons/skip.svg'
export {default as skipFilled} from './icons/skipFilled.svg'
export {default as sort} from './icons/sort.svg'
export {default as star} from './icons/star.svg'
export {default as starFilled} from './icons/starFilled.svg'
export {default as store} from './icons/store.svg'
export {default as storeFilled} from './icons/storeFilled.svg'
export {default as sunset} from './icons/sunset.svg'
export {default as sunsetFilled} from './icons/sunsetFilled.svg'
export {default as thumbsDown} from './icons/thumbsDown.svg'
export {default as thumbsUp} from './icons/thumbsUp.svg'
export {default as ticket} from './icons/ticket.svg'
export {default as ticketFilled} from './icons/ticketFilled.svg'
export {default as tobacco} from './icons/tobacco.svg'
export {default as twitter} from './icons/twitter.svg'
export {default as unlock} from './icons/unlock.svg'
export {default as unlockFilled} from './icons/unlockFilled.svg'
export {default as view} from './icons/view.svg'
export {default as viewFilled} from './icons/viewFilled.svg'
export {default as warning} from './icons/warning.svg'
export {default as warningFilled} from './icons/warningFilled.svg'
export {default as weight} from './icons/weight.svg'
export {default as weightFilled} from './icons/weightFilled.svg'
export {default as x} from './icons/x.svg'
export {default as xCircle} from './icons/xCircle.svg'
export {default as xCircleFilled} from './icons/xCircleFilled.svg'
export {default as xSmall} from './icons/xSmall.svg'
export {default as zoomIn} from './icons/zoomIn.svg'
export {default as zoomOut} from './icons/zoomOut.svg'
