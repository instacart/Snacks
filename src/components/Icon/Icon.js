import React from 'react'
import PropTypes from 'prop-types'
import hexValues from './hexValues'

const baseStyles = {
  fontSize: '16px',
  position: 'relative',
  fontFamily: 'ic-icons',
  fontStyle: 'normal',
  fontWeight: 'normal',
  fontVariant: 'normal',
  textTransform: 'none',
  speak: 'none',
  lineHeight: '1',
  fontSmoothing: 'antialiased',
  osxFontSmoothing: 'grayscale',
}

const getIcon = ({ name, code }) => {
  const iconCode = !code ? hexValues[name] : code
  const codePoint = parseInt(iconCode, 16)
  return String.fromCodePoint(codePoint)
}

const Icon = props => {
  const { className, style, onClick } = props
  const icon = getIcon(props)
  return (
    <i className={className} css={[baseStyles, style]} aria-hidden onClick={onClick}>
      {icon}
    </i>
  )
}

Icon.propTypes = {
  className: PropTypes.string,
  /**
   * String name of icon - ex 'cart'
   */
  name: PropTypes.oneOf(Object.keys(hexValues)),
  /** Hexcode of desired icon from ic-icons */
  code: PropTypes.string,
  /**
   * Optional style overrides merged into emotion css
   *
   * @deprecated
   * This prop exists for backwards compatibility and will be
   * removed in a future version
   */
  style: PropTypes.object,
  /** Callback function called after button click
   * @param {SyntheticEvent} event The react `SyntheticEvent`
   */
  onClick: PropTypes.func,
}

export default Icon
