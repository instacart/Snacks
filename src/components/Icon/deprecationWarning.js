import svgIcons from '../SVGIcon/icons'
import hexValues from './hexValues'

const hexLookup = Object.keys(hexValues).reduce((lookup, value) => (
  { ...lookup, [hexValues[value]]: value }
), {})

const renamed = {
  arrowDownSmall: 'arrowDown', arrowDownSmallBold: 'arrowDownSmall', arrowLeftSmall: 'arrowLeft', arrowLeftSmallBold: 'arrowLeftSmall', arrowRightSmall: 'arrowRight', arrowRightSmallBold: 'arrowRightSmall', arrowUpSmall: 'arrowUp', arrowUpSmallBold: 'arrowUpSmall', bogo: 'lightning', bogoFilled: 'lightningFilled', check: 'checkmark', checkProgressComplete: 'checkmarkCircle', checkProgressDashed: 'checkmarkCircleDashed', deals: 'coupons', dealsFilled: 'couponsFilled', facebookFilled: 'facebook', hamburger: 'menu', happy: 'reaction', iconPerson: 'person', iconPersonAdd: 'personAdd', iconPersonAddFilled: 'personAddFilled', iconPersonFilled: 'personFilled', listsAndRecipes: 'recipes', message: 'chat', messageFilled: 'chatFilled', note: 'instructions', noteFilled: 'instructionsFilled', orderHistory: 'orders', orderProblem: 'orderIssue', phone: 'call', phoneFilled: 'callFilled', phoneIphone: 'phone', phoneIphoneFilled: 'phoneFilled', picture: 'photo', pinterestFilled: 'pinterest', pricing: 'sale', pricingFilled: 'saleFilled', reccuring: 'recurringOrder', replace: 'replacement', save: 'saveForLater', saveFilled: 'saveForLaterFilled', specials: 'glitter', specialsFilled: 'glitterFilled', trash: 'delete', trashFilled: 'deleteFilled', twitterFilled: 'twitter', xBold: 'xSmall',
  checkBold: 'checkmark', iconExit: 'logout', minusBold: 'minus', plusBold: 'plus', pricingSameFilled: 'checkmarkCircleFilled', replaceBold: 'replace',
}

export default function deprecationWarning({ name, code, style }) {
  if(!window.ICSnacksDeprecationWarnings) return
  if(!window.ICSnacksDeprecationWarnings.SVGIcon) return
  const messages = ['The `Icon` component has been deprecated.']
  messages.push('Please use the `SVGIcon` component instead.')
  if(code) {
    messages.push('The `code` prop has also been depreacted, please use `name` instead.')
    name = hexLookup[code]
  }
  if(name in renamed) {
    messages.push(`The icon name '${name}' has been renamed to '${renamed[name]}'.`)
  } else if(!(name in svgIcons)) {
    messages.push(`The icon name '${name}' has been removed.`)
  }
  if(style) {
    const styleCopy = { ...style }
    if(style.fontSize) {
      delete styleCopy.fontSize
      messages.push('Sizing the Icon component with the `fontSize` style property is deprecated, use the `size` prop instead.')
    }
    if(style.color) {
      delete styleCopy.color
      messages.push('Coloring the Icon component with the `color` style property is deprecated, use the `color` prop instead.')
    }
    const styleKeys = Object.keys(styleCopy)
    if(styleKeys.length > 0) {
      messages.push(`When converting to 'SVGIcon', styling will be different please audit the following style properties and verify they will work with SVGIcon: ${styleKeys.join(', ')}`)
    }
  }
  console.warn(...messages)
}
