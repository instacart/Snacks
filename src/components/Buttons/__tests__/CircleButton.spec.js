import React from 'react'
import renderer from 'react-test-renderer'
import { StyleRoot } from 'radium'
import { mount } from 'enzyme'
import { spy } from 'sinon'
import Icon from '../../Icon/Icon'
import CircleButton from '../CircleButton'

it('renders basic CircleButton correctly', () => {
  const tree = renderer
    .create(
      <StyleRoot>
        <div>
          <CircleButton>1</CircleButton>
        </div>
      </StyleRoot>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders CircleButton with Icon correctly', () => {
  const tree = renderer
    .create(
      <StyleRoot>
        <div>
          <CircleButton>
            <Icon name="arrowLeftSmallBold" />
          </CircleButton>
        </div>
      </StyleRoot>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders CircleButton with passed style prop correctly', () => {
  const tree = renderer
    .create(
      <StyleRoot>
        <div>
          <CircleButton
            style={{
              top: '8px',
              right: '8px',
              position: 'absolute',
            }}
          >
            1
          </CircleButton>
        </div>
      </StyleRoot>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders CircleButton with passed ariaLabel correctly', () => {
  const tree = renderer
    .create(
      <StyleRoot>
        <div>
          <CircleButton ariaLabel={'this is a button test'}>1</CircleButton>
        </div>
      </StyleRoot>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders CircleButton with passed disabled prop correctly', () => {
  const tree = renderer
    .create(
      <StyleRoot>
        <div>
          <CircleButton disabled>1</CircleButton>
        </div>
      </StyleRoot>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('calls onClick callback correctly', () => {
  const onClick = spy()
  const CButton = mount(
    <StyleRoot>
      <div>
        <CircleButton onClick={onClick}>1</CircleButton>
      </div>
    </StyleRoot>
  )

  const button = CButton.find('button').first()
  button.simulate('click')
  expect(onClick.calledOnce).toBeTruthy()
})

it('does not call onClick when disabled prop is true', () => {
  const onClick = spy()
  const CButton = mount(
    <StyleRoot>
      <div>
        <CircleButton onClick={onClick} disabled>
          1
        </CircleButton>
      </div>
    </StyleRoot>
  )

  const button = CButton.find('button').first()
  button.simulate('click')
  expect(onClick.calledOnce).toBeFalsy()
})
