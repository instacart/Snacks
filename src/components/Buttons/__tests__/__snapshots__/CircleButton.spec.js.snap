// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CircleButton with Icon correctly 1`] = `
.emotion-1 {
  z-index: 100;
  border: 0;
  width: 48px;
  height: 48px;
  border-radius: 50%;
  box-shadow: 0 1px 2px 0 rgba(0,0,0,0.26),0 1px 4px 0 rgba(0,0,0,0.16);
  -webkit-transition: background-color 150ms ease-in-out;
  transition: background-color 150ms ease-in-out;
  text-align: center;
  line-height: 1;
  background-color: #43B02A;
  color: #FFFFFF;
}

@media (max-width:767px) {
  .emotion-1 {
    display: none;
  }
}

.emotion-1:hover {
  background-color: #177D05;
}

.emotion-1:active {
  background-color: #177D05;
}

.emotion-1:focus {
  background-color: #177D05;
}

.emotion-0 {
  font-size: 16px;
  position: relative;
  font-family: ic-icons;
  font-style: normal;
  font-weight: normal;
  font-variant: normal;
  text-transform: none;
  speak: none;
  line-height: 1;
  font-smoothing: antialiased;
  osx-font-smoothing: grayscale;
}

<div>
  <button
    className="emotion-1"
    onClick={[Function]}
  >
    <i
      aria-hidden={true}
      className="emotion-0"
    >
      î€‘
    </i>
  </button>
</div>
`;

exports[`renders CircleButton with passed ariaLabel correctly 1`] = `
.emotion-0 {
  z-index: 100;
  border: 0;
  width: 48px;
  height: 48px;
  border-radius: 50%;
  box-shadow: 0 1px 2px 0 rgba(0,0,0,0.26),0 1px 4px 0 rgba(0,0,0,0.16);
  -webkit-transition: background-color 150ms ease-in-out;
  transition: background-color 150ms ease-in-out;
  text-align: center;
  line-height: 1;
  background-color: #43B02A;
  color: #FFFFFF;
}

@media (max-width:767px) {
  .emotion-0 {
    display: none;
  }
}

.emotion-0:hover {
  background-color: #177D05;
}

.emotion-0:active {
  background-color: #177D05;
}

.emotion-0:focus {
  background-color: #177D05;
}

<div>
  <button
    aria-label="this is a button test"
    className="emotion-0"
    onClick={[Function]}
  >
    1
  </button>
</div>
`;

exports[`renders CircleButton with passed disabled prop correctly 1`] = `
.emotion-0 {
  z-index: 100;
  border: 0;
  width: 48px;
  height: 48px;
  border-radius: 50%;
  box-shadow: 0 1px 2px 0 rgba(0,0,0,0.26),0 1px 4px 0 rgba(0,0,0,0.16);
  -webkit-transition: background-color 150ms ease-in-out;
  transition: background-color 150ms ease-in-out;
  text-align: center;
  line-height: 1;
  background-color: #43B02A;
  color: #FFFFFF;
}

@media (max-width:767px) {
  .emotion-0 {
    display: none;
  }
}

.emotion-0:hover {
  background-color: #177D05;
}

.emotion-0:active {
  background-color: #177D05;
}

.emotion-0:focus {
  background-color: #177D05;
}

<div>
  <button
    className="emotion-0"
    disabled={true}
    onClick={[Function]}
  >
    1
  </button>
</div>
`;

exports[`renders CircleButton with passed style prop correctly 1`] = `
.emotion-0 {
  z-index: 100;
  border: 0;
  width: 48px;
  height: 48px;
  border-radius: 50%;
  box-shadow: 0 1px 2px 0 rgba(0,0,0,0.26),0 1px 4px 0 rgba(0,0,0,0.16);
  -webkit-transition: background-color 150ms ease-in-out;
  transition: background-color 150ms ease-in-out;
  text-align: center;
  line-height: 1;
  background-color: #43B02A;
  color: #FFFFFF;
  top: 8px;
  right: 8px;
  position: absolute;
}

@media (max-width:767px) {
  .emotion-0 {
    display: none;
  }
}

.emotion-0:hover {
  background-color: #177D05;
}

.emotion-0:active {
  background-color: #177D05;
}

.emotion-0:focus {
  background-color: #177D05;
}

<div>
  <button
    className="emotion-0"
    onClick={[Function]}
  >
    1
  </button>
</div>
`;

exports[`renders basic CircleButton correctly 1`] = `
.emotion-0 {
  z-index: 100;
  border: 0;
  width: 48px;
  height: 48px;
  border-radius: 50%;
  box-shadow: 0 1px 2px 0 rgba(0,0,0,0.26),0 1px 4px 0 rgba(0,0,0,0.16);
  -webkit-transition: background-color 150ms ease-in-out;
  transition: background-color 150ms ease-in-out;
  text-align: center;
  line-height: 1;
  background-color: #43B02A;
  color: #FFFFFF;
}

@media (max-width:767px) {
  .emotion-0 {
    display: none;
  }
}

.emotion-0:hover {
  background-color: #177D05;
}

.emotion-0:active {
  background-color: #177D05;
}

.emotion-0:focus {
  background-color: #177D05;
}

<div>
  <button
    className="emotion-0"
    onClick={[Function]}
  >
    1
  </button>
</div>
`;
